### ISSUE TITLE
Support Gateway API spec

### ISSUE THREAD

**Issue by lucasfcnunes** [loft-sh/vcluster#2142]
### Is your feature request related to a problem?

The Gateway API is in v1 and is supposed to be able to completely replace the usage of Ingress and give more possibilities, power, expressiveness and security to users.

> Ingress is frozen. New features are being added to the [Gateway API](https://kubernetes.io/docs/concepts/services-networking/gateway/). - [source](https://kubernetes.io/docs/concepts/services-networking/ingress/#:~:text=Ingress%20is%20frozen.%20New%20features%20are%20being%20added%20to%20the%20Gateway%20API.)

> Gateway API is the successor to the [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) API. However, it does not include the Ingress kind. As a result, a one-time conversion from your existing Ingress resources to Gateway API resources is necessary.
Refer to the [ingress migration](https://gateway-api.sigs.k8s.io/guides/migrating-from-ingress/#migrating-from-ingress) guide for details on migrating Ingress resources to Gateway API resources. - [source](https://kubernetes.io/docs/concepts/services-networking/gateway/#migrating-from-ingress:~:text=Gateway%20API%20is%20the,to%20Gateway%20API%20resources.)

[Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/):
![image](https://github.com/user-attachments/assets/8cbcd460-d335-4e41-8489-10339fe4bb2c)

[Gateway API](https://kubernetes.io/docs/concepts/services-networking/gateway/):

- HTTP flow
![image](https://github.com/user-attachments/assets/3131206f-ac32-475b-bd2c-269699074eab)

- Users (Personas)
![image](https://github.com/user-attachments/assets/f1e27842-47ab-43b0-80a4-820965037842)

### Which solution do you suggest?

Have for vcluster...

- full compliance with https://gateway-api.sigs.k8s.io/reference/spec/
- docs for how to manage and use shared (host x vcluster) Gateway API resources
- tested GatewayClass (like in [e2e_generic/clusterscope/ingressclasses.go](https://github.com/loft-sh/vcluster/blob/2d00aafe1541aa6c66a414e33780353a00814456/test/e2e_generic/clusterscope/ingressclasses.go)) and related
- listed caveats when using Gateway API

### Which alternative solutions exist?

NA

### Additional context

- Many projects have already reported conformance https://gateway-api.sigs.k8s.io/implementations/v1.1/
- https://github.com/kubernetes/ingress-nginx is going to implement it too https://youtube.com/clip/UgkxgCwz2N3y19vEyQh-vBYHR78ekQJnDtfA?si=8dSf3JK4pU6xTz8Z

**Comment by BonzTM** [comment]
What is everybody doing about GWAPI and vCluster in the meantime?  GA was in 2023 and 1.2 is out now.  Users are deploying it.

**Comment by poblin-orange** [comment]
Ingress Nginx behind deprecated, time to move to Gateway API for many of your users. Any feedback or roadmap ?

**Comment by EliSauder** [comment]
As apart of the gateway resources, will there be support for syncing the routes from the vcluster similar to how ingresses are synced?

**Comment by cbron** [comment]
Yes, we would support syncing gateway CRDs like HTTPRoute to the host. We'll be starting development on this project soon.

**Comment by djas19** [comment]
Is there an update on when we can expect this? It's a blocker for us to adopt vclusters at the moment

**Comment by Netroxen** [comment]
bumping this as well, the documentation mentions customResources syncing is "FREE", however even after upgrading to the latest platform version, I still receive the "you are trying to use a vCluster pro feature". This is seriously hindering the use of vCluster, moreover because of the GatewayAPI ...

**Comment by matskiv** [comment]
I think this was already released in v0.35, as per [changelog](https://loft.featurebase.app/changelog/vcluster-platform-v410-vcluster-v035-deepened-argo-cd-integration-virtual) and here are docs:
https://www.vcluster.com/docs/vcluster/configure/vcluster-yaml/sync/to-host/networking/gateway-api
https://www.vcluster.com/docs/vcluster/configure/vcluster-yaml/sync/from-host/gateways

**Comment by djas19** [comment]
I just updated my stack and it looks like manifests are now synced correctly from and to the host on v0.35.1
